### PR TITLE
WIP - GAS Golfing

### PR DESCRIPTION
**Pull request by jamesmorgan**

### CODE REVIEW

File: contracts/marketplace/KODAV3GatedMerkleMarketplace.sol [hunk 1]
```diff
@@ -92,7 +92,7 @@ abstract contract KODAV3GatedMerkleMarketplace is BaseUpgradableMarketplace, IKO
         uint256 editionId = saleToEditionId[_saleId];
         require(editionId > 0, "no sale exists");
         require(!isSalePaused[_saleId], 'sale is paused');
-        require(!koda.isEditionSoldOut(editionId), 'the sale is sold out');
+//        require(!koda.isEditionSoldOut(editionId), 'the sale is sold out');
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3GatedMerkleMarketplace.sol:95]: Kill extra check

File: contracts/marketplace/KODAV3GatedMerkleMarketplace.sol [hunk 2]
```diff
@@ -207,6 +207,6 @@ abstract contract KODAV3GatedMerkleMarketplace is BaseUpgradableMarketplace, IKO
         );
     }
 
-    function _handleEditionSaleFunds(uint256 _saleId, uint256 _editionId, address _receiver) internal virtual;
+    function _handleEditionSaleFunds(uint256 _saleId, uint256 _editionId) internal virtual;
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3GatedMerkleMarketplace.sol:210]: No need to pass through the param

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 3]
```diff
@@ -23,9 +23,11 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
     /// @notice emitted when a phase is removed from a sale
     event PhaseRemoved(uint256 indexed saleId, uint256 indexed editionId, uint256 indexed phaseId);
 
-    // TODO do we need mintCount?
     /// @notice emitted when someone mints from a sale
-    event MintFromSale(uint256 saleId, uint256 tokenId, uint256 phaseId, address account, uint256 mintCount);
+    event MintFromSale(uint256 saleId, uint256 tokenId, uint256 phaseId, address account);
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:27]: Chop mint count - not needed in the event

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 4]
```diff
@@ -45,16 +47,18 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
         uint128 endTime; // The end time of the sale phase, also the beginning of the next phase if applicable
         uint16 walletMintLimit; // The mint limit per wallet for the phase
         uint128 priceInWei; // Price in wei for one mint
-        bytes32 merkleRoot; // The merkle tree root for the phase
-        string merkleIPFSHash; // The IPFS hash referencing the merkle tree
         uint128 mintCap; // The maximum amount of mints for the phase
         uint128 mintCounter; // The current amount of items minted
+        bytes32 merkleRoot; // The merkle tree root for the phase
+        string merkleIPFSHash; // The IPFS hash referencing the merkle tree
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:53]: Struct reorder

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 5]
```diff
@@ -45,16 +47,18 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
...
     }
 
     /// @notice Sale represents a gated sale, with mapping links to different sale phases
     struct Sale {
         uint256 id; // The ID of the sale
         uint256 editionId; // The ID of the edition the sale will mint
+        address creator;
+        address fundsReceiver;
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:61]: store these on creation to define the creator and receiver? may need some validation, thoughts?

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 6]
```diff
@@ -75,7 +80,8 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
 
     modifier onlyCreatorOrAdmin(uint256 _editionId) {
         require(
-            accessControls.hasAdminRole(_msgSender()) || koda.getCreatorOfEdition(_editionId) == _msgSender(),
+        // TODO quick exit to save GAS
+            koda.getCreatorOfEdition(_editionId) == _msgSender() || accessControls.hasAdminRole(_msgSender()),
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:84]: quick exit here, assume its the creator not admin

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 7]
```diff
@@ -189,12 +204,13 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
     }
 
     /// @dev checks whether a given user is on the list to mint from a phase
-    function onPhaseMintList(uint256 _saleId, uint _phaseId, uint256 _index, address _account, bytes32[] calldata _merkleProof)
+    function onPhaseMintList(uint256 _saleId, uint256 _phaseId, uint256 _index, address _account, bytes32[] calldata _merkleProof)
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:207]: method param reorder compacting args

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 8]
```diff
@@ -241,39 +257,47 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
 
     function _handleMint(uint256 _saleId, uint256 _phaseId, uint256 _editionId, uint16 _mintCount, address _recipient) internal override {
         require(_mintCount > 0, "Nothing being minted");
-        address _receiver;
+
+        address creator = sales[_saleId].creator;
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:261]: Lookup stored creator

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 9]
```diff
@@ -241,39 +257,47 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
...
+    function _handleEditionSaleFunds(uint256 _saleId, uint256 _editionId) internal override {
         uint256 platformPrimarySaleCommission = _getPlatformSaleCommissionForSale(_saleId);
         uint256 koCommission = (msg.value / modulo) * platformPrimarySaleCommission;
         if (koCommission > 0) {
             (bool koCommissionSuccess,) = platformAccount.call{value : koCommission}("");
             require(koCommissionSuccess, "commission payment failed");
         }
 
-        (bool success,) = _receiver.call{value : msg.value - koCommission}("");
+        (bool success,) = sales[_saleId].fundsReceiver.call{value : msg.value - koCommission}("");
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:292]: lookup and dont pass through

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 10]
```diff
@@ -286,19 +310,18 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
...
-        uint256 saleId = saleIdCounter;
-        uint256 editionSize = koda.getSizeOfEdition(sales[saleId].editionId);
+        uint256 editionSize = koda.getSizeOfEdition(sales[saleIdCounter].editionId);
         require(editionSize > 0, 'edition does not exist');
 
         uint256 numOfPhases = _startTimes.length;
         for (uint256 i; i < numOfPhases; ++i) {
             require(_endTimes[i] > _startTimes[i], 'phase end time must be after start time');
-            require(_walletMintLimits[i] > 0 && _walletMintLimits[i] < editionSize, 'phase mint limit must be greater than 0');
+            require(_walletMintLimits[i] > 0 && _walletMintLimits[i] <= editionSize, 'phase mint limit must be greater than 0');
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:319]: bug

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol [hunk 11]
```diff
@@ -286,19 +310,18 @@ contract KODAV3UpgradableGatedMarketplace is BaseUpgradableMarketplace, KODAV3Ga
         uint128[] memory _pricesInWei,
         uint128[] memory _mintCaps
     ) internal {
-        uint256 saleId = saleIdCounter;
```
Comment (jamesmorgan) [inline contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:289]: unnecessary variable

File: contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol (hunk 5 above)
Comment (andygray) [reply contracts/marketplace/KODAV3UpgradableGatedMarketplace.sol:61]: could go in the tree in theory

**Pull request closed without merging**